lib/options.nix

```diff
@@ -243,6 +361,35 @@ rec {
     apply = x: throw "Option value is not readable because the option is not declared.";
   } // attrs);
 
+  /**
+    Merges multiple definitions of an option into a single value
+    by checking that they are all equal.
+
+    # Inputs
+
+    `loc`
+    : location of the option in the configuration as a list of strings.
+
+      e.g. ["boot" "loader "grub" "enable"]
+
+    `defs`
+    : list of definition values and locations.
+
+      e.g. [ { file = "/foo.nix"; value = 1; } { file = "/bar.nix"; value = 2 } ]
```

Comment:
This should simply refer to `mkOptionType`. That's where all the merge functions are passed to, so it's the best place to describe the type of those functions.
Implementations don't have to repeat that, because they aren't actually called by users.

Reply:
Yep had the same though. Gona fix the example.

Reply:
Just took a look into the nixpkgs manual. It seems we don't explain the concept of option types. At all....